app/src/main/java/com/cornellappdev/transit/ui/screens/HomeScreen.kt

```diff
@@ -0,0 +1,153 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.cornellappdev.transit.R
+import com.cornellappdev.transit.ui.viewmodels.HomeViewModel
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.isGranted
+import com.google.android.gms.maps.model.CameraPosition
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.MapProperties
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.MarkerState
+import com.google.maps.android.compose.rememberCameraPositionState
+import com.google.accompanist.permissions.rememberPermissionState
+import androidx.lifecycle.viewmodel.compose.viewModel
+
+@OptIn(ExperimentalPermissionsApi::class, ExperimentalMaterial3Api::class)
+@Composable
+fun HomeScreen(
+    homeViewModel: HomeViewModel = viewModel()
+) {
+
+    /**
+     * Permissions dialog
+     */
+    val permissionState = rememberPermissionState(android.Manifest.permission.ACCESS_FINE_LOCATION)
+    val openDialog = remember { mutableStateOf(true) }
+
+    if (openDialog.value && !permissionState.status.isGranted) {
+        AlertDialog(
+            onDismissRequest = {
+
+                openDialog.value = false
+            }
+        ) {
+            Surface(
+                modifier = Modifier
+                    .wrapContentWidth()
+                    .wrapContentHeight(),
+                shape = RoundedCornerShape(80.dp),
+                tonalElevation = AlertDialogDefaults.TonalElevation
+            ) {
+                Column(modifier = Modifier.padding(40.dp)) {
+                    Text(
+                        text = stringResource(R.string.permission_dialog_text)
+                    )
+                    Spacer(modifier = Modifier.height(30.dp))
+                    TextButton(
+                        onClick = {
+                            permissionState.launchPermissionRequest()
+                            openDialog.value = false;
+                        },
+                        modifier = Modifier.align(Alignment.CenterHorizontally)
+                    ) {
+                        Text("Allow")
+                    }
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Google Maps and Search
+     */
+
+    //Map state
+    val defaultIthaca = LatLng(42.44, -76.50)
+    val cameraPositionState = rememberCameraPositionState {
+        position = CameraPosition.fromLatLngZoom(defaultIthaca, 12f)
+    }
+
+    // Search state
+    var query by remember { mutableStateOf("") }
+    var searchActive by remember { mutableStateOf(false) }
```

Comment:
Consider moving these bits of info to the `ViewModel`. I think information like this is really important for our data flow, so I think it'd be best to move these into a state flowing down the viewmodel.

Seems like you've put this as a TODO for now, so LMK if you're just waiting to do this.